controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -26,8 +26,8 @@ import (
 const (
 	RKE2ControlPlaneFinalizer = "rke2.controleplane.cluster.x-k8s.io"
 
-	// RKE2ServerConfigurationAnnotation is a machine annotation that stores the json-marshalled string of KCP ClusterConfiguration.
-	// This annotation is used to detect any changes in ClusterConfiguration and trigger machine rollout in KCP.
+	// RKE2ServerConfigurationAnnotation is a machine annotation that stores the json-marshalled string of RKE2Config
```

Comment:
Do we use this annotation anywhere?

Reply:
This annotation is used for future upgrade functionality. The logic is copied over from Kubeadm and is implemented in : 
`controlplane.MachinesNeedingRollout` -> `controlplane.MatchesRCPConfiguration` -> `controlplane.MatchesRKE2BootstrapConfig` . The latter is not yet implemented, but it should use the above annotation and compare it to the current `RKE2ControlPlane.AgentConfig` value.